Feature/profile health data

CalebHasch opened the pull request:
## Describe your changes
Create a component that shows health info and can open a form to update it.

## Issue ticket number and link
ticket#: 50
link: https://tripleten-apiary.atlassian.net/jira/software/projects/DROP/boards/5?selectedIssue=DROP-50

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [x] I have made any necessary updates to Storybook to accompany these changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

GodisNanu commented:
I see you made some changes to a component.  I think you are supposed to update storybook whenever you do that. I also wanted to know why some form inputs aren't set as required.  You only set the sex input to required, but it seems like the others should be required as well.

CalebHasch commented:
> I see you made some changes to a component. I think you are supposed to update storybook whenever you do that. I also wanted to know why some form inputs aren't set as required. You only set the sex input to required, but it seems like the others should be required as well.

good catches. I updated the selectForm story and relooking at the backend schema sex should not be required, only the diabetes type so I fixed that. Should be good to go now unless you see something else.

Aivaxela commented on src/routes/settings/sections/health-data-form.tsx:
```diff
@@ -0,0 +1,169 @@
+  }
+
+   // Process dateOfBirth (if provided)
+   if (dateOfBirth.trim()) {
```
We do have a datepicker form element....but it isn't very easy to select the year without click the "back" button a whole bunch...What you have is fine I just wanted to mention that, I feel like we need a better date picker eventually for instances like this.

Aivaxela reviewed (commented):
Nice work, looks great and it seems like the queries are working great!
Left a few comments of some minor things.

CalebHasch commented on src/routes/settings/sections/health-data-form.tsx:
```diff
@@ -0,0 +1,169 @@
+      updateData.height = feet * 12 + inches;
+      clearErrors("height"); // Clear error if the format is correct
+    } else {
+      setError("height", { type: "manual", message: "Invalid height format. Use ft'in or ft'in\"" });
```
I was trying to give them the option of adding the inch quote so they could put 5'12" or 5'12 and I would format it on my end to correct it, but I do think some people could find the validation confusing. I really like your dropdown idea so I'll ask Josh if I can implement it because I do think that would be an easier user experience.

CalebHasch replied:
Josh thought it was a good idea so I went ahead and implemented the dropdowns

CalebHasch commented on src/routes/settings/sections/health-data-form.tsx:
```diff
@@ -0,0 +1,169 @@
+  }
+
+  try {
+    const response = await handleUpdate("paul@gmail.com", updateData);
```
oh yep, that's exactly what happed. I should get in the habit of add a "For testing" comment about it so I remember to change it back later. It should use the current users email now correctly now.

Aivaxela reviewed (approved):
LGTM
